responsemanager/server.go

```diff
@@ -239,10 +289,13 @@ func (rm *ResponseManager) finishTask(task *peertask.Task, err error) {
 	log.Infow("graphsync response processing complete (messages stil sending)", "request id", key.requestID, "peer", key.p, "total time", time.Since(response.startTime))
 
 	if err != nil {
+		response.span.RecordError(err)
+		response.span.SetStatus(codes.Error, err.Error())
 		log.Infof("response failed: %w", err)
 	}
 	delete(rm.inProgressResponses, key)
 	response.cancelFn()
+	response.span.End()
```

Comment:
reminder: this is not where the request actually ends -- see https://github.com/ipfs/go-graphsync/issues/288